File: 0554_brickWall/main.cpp
#include <vector>
#include <unordered_map>
#include <algorithm>

class Solution {
public:
    int leastBricks(std::vector<std::vector<int>>& wall) {
        std::unordered_map<int,int> map;
        int n = wall.size();
        for(int i = 0; i < n; ++i){
            int sum=0;
            for(int j = 0; j < wall[i].size()-1; ++j){
                sum += wall[i][j];
                ++map[sum];
            }
        }

        int maxCount=0;
        for(auto m: map){
            if(m.second > maxCount)
                maxCount = m.second;
        }
        return n - maxCount;
    }
};

File: 0554_brickWall/main.py
from typing import List
from collections import defaultdict

class Solution:
    def leastBricks(self, wall: List[List[int]]) -> int:
        myMap = defaultdict(int)

        for rowList in wall:
            pos = 0
            for val in rowList[:-1]:
                pos += val
                myMap[pos] += 1
        
        maxCount = 0
        for _, count in myMap.items():
            maxCount = max(maxCount, count)
        res = len(wall) - maxCount
        return res